sources/engine/Stride.Audio/Layers/XAudio/AudioProvider.cs

```diff
@@ -26,8 +27,12 @@ public AudioProvider()
 
         public unsafe AudioBuffer BufferCreate(int maxBufferSizeBytes)
         {
-            var buffer = new AudioBuffer(maxBufferSizeBytes);
-			//buffer.buffer.PContext = buffer;
+            var buffer = new AudioBuffer();
+            AudioBuffer* bufferPtr = &buffer;        
+			buffer.Buffer.PContext = bufferPtr;
+            
+            var data = stackalloc byte[maxBufferSizeBytes];
+			buffer.Buffer.PAudioData = data;
 			return buffer;
```

Comment:
Excuse me if I speak prematurely, but I stumble upon this and it hurts my eyes. It is never valid to `stackalloc` and let the pointer escape the current method. `data` is implicitly discarded and will be reused for method calls immediately upon return from this method. This buffer should probably be a rented buffer and perhaps also allocated from the pinned object heap, or allocated using `NativeMemory`, since I assume it will be used for native interop and/or hardware.